Kickstarter-iOS/Features/MessageThreads/Controller/MessageThreadsViewController.swift

```diff
@@ -85,6 +85,9 @@ internal final class MessageThreadsViewController: UITableViewController {
   ) {
     if let messageThread = self.dataSource[indexPath] as? MessageThread {
       let vc = MessagesViewController.configuredWith(messageThread: messageThread)
+
+      vc.dismissedAfterBlockingUserHandler = { self.viewModel.inputs.refresh() }
```

Comment:
I'm concerned that this could cause a retain loop. The messagesViewController will now hold a strong reference to the messageThreadViewController. I don't think there's currently a strong reference in the opposite direction, but I'm not 100% sure how the navigation controller works, and it wouldn't surprise me if this gets refactored at some point in the future so that the messageThreadViewController does hold on to the messagesViewController. How would you feel about using weak self in the block or alternatively adding a comment that calls out why this is currently safe (or why it might not be safe in the future)?

Reply:
It's a good point. I considered this while testing and couldn't see evidence of a retain cycle. As an extra precaution, I'll add a capture list and trigger this callback just before the child VC gets dismissed. That's a good call out.